mlir/utils/jenkins/Jenkinsfile

```diff
@@ -793,5 +818,78 @@ pipeline {
                 }
             }
         }
+        stage ("MIGraphX") {
+            when {
+                beforeAgent true;
+                anyOf {
+                    equals expected: true, actual: params.checkMIGraphX;
+                    equals expected: true, actual: params.nightly;
```

Comment:
My opinion is to run:
 - Minimal test using migraphx driver on PR CI (1 config per direction, per data type)
 - Resnet50 test in weekly CI

I don't think nightly CI is the right place for it (Unless we want to measure performance impact). 

This is just so that there is 1:1 correspondence between MIOpen and MIGraphX integration.

Reply:
We do run Resnet50 e2e tests (in tosa & migraphx dialects) in a nightly basis and they are not that slow to be pushed to be weekly (as compared to tuning). The e2e tests are more functional in nature that captures everything is correct (migraphx passes that happens before MLIR offloading and their compatibility of the MLIR offload done later in their pipeline) -- so unless there is a concern on time increase , I would prefer it to be nightly.

MIGraphx is inference only AFAIK, hence no directions. I feel extending the testing on MIGraphX is task on its own.
I do not intend extend the existing unit tests in the scope of the PR to keep this as basic minimal PR to get the CI bits hooked up. We can do follow up on MIGraphX side (that owns the unit tests for MLIR) to put more things there, if needed. The scope of this PR is just to make sure whatever tests and functionality that exists and tested are not broken by development in MLIR.

Reply:
However, we could run the unit tests on the PR basis but our prior discussion was to MIGraphX testing on PR that we suspect to break things (e.g. fusion).

Reply:
Thanks for giving detailed reply on your opinions! I don't mean to block this PR with the comment. What you have on nightly with very limited impact on its runtime sounds reasonable. 

PR wise, speaking of my experience from MIOpen side, I think I still prefer some minimal coverage to make sure that things still build and work. There are a dozen times that the MIOpen minimal coverage PR testing captured a non-trivial kernel bug and help us to notice that early.

So to rephrase, I think what you have is good. We can have follow-ups on what make sense to track on PR.

Reply:
No worries! :) 

I dont disagree with you; and more than happy to promote things to PR CI if things are falling between the cracks (i.e. seeing our nightlies fail because we are not enforcing checkMIGraphX option on relevant PR as we thought we would be).